File: ceropath/model/sequence.py
from core import Core
from individual import Individual
from gene import Gene
from primer import Primer

class Sequence(Core):
    collection_name = "sequence"
    structure = {
        'individual': Individual,
        'gene': Gene,
        'sequence': unicode,
        'operator': unicode,
        'primer':{
            'forward': Primer,
            'reverse': Primer,
        },
        'chromatogram_link': unicode,
        'length': int,
        'accession_number': unicode,
        'internet_display': bool,
    }
    use_autorefs = True
    indexes = [
        {'fields':['individual', 'gene'], 'unique':True},
    ]

File: config/schemas/sequence.js

module.exports = {
    properties: {
        title: {
            type: 'string'
        },
        individualID: {
            type: 'Individual',
            label: 'individual'
        },
        gene: {
            type: 'string'
        },
        operator: {
            type: 'string'
        },
        forwardPrimer: {
            type: 'Primer'
        },
        reversePrimer: {
            type: 'Primer'
        },
        sequence: {
            type: 'string'
        }
    }
};
